state/state_test.go

```diff
@@ -179,8 +179,16 @@ func (s *StateSuite) TestParseIDToTag(c *gc.C) {
 	unTag := state.ParseLocalIDToTags(unit)
 	apTag := state.ParseLocalIDToTags(application)
 
-	c.Assert(moTag.String(), gc.Equals, "model-e")
-	c.Assert(maTag.String(), gc.Equals, "machine-m#0")
+	_, err := names.ParseTag(moTag.String())
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = names.ParseTag(maTag.String())
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = names.ParseTag(unTag.String())
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = names.ParseTag(apTag.String())
+
+	c.Assert(moTag.String(), gc.Equals, "model-42c4f770-86ed-4fcc-8e39-697063d082bc:e")
```

Comment:
Alternatively, one could use the above tag and compare them instead. 
This is more clear, the other way would be a more thorough test in serialization and deserialization of names.